internal/chart/flags.go

```diff
@@ -13,59 +18,48 @@ type FlagsBuilder interface {
 }
 
 type flagsBuilder struct {
-	dockerRegistryChartParams map[string]interface{}
-	dockerRegistryParams      map[string]interface{}
-	containersDataParams      map[string]interface{}
-	webhookValuesParams       map[string]interface{}
-	globalParams              map[string]interface{}
+	flags map[string]interface{}
 }
 
 func NewFlagsBuilder() FlagsBuilder {
 	return &flagsBuilder{
-		dockerRegistryChartParams: map[string]interface{}{},
-		dockerRegistryParams:      map[string]interface{}{},
-		containersDataParams:      map[string]interface{}{},
-		webhookValuesParams:       map[string]interface{}{},
-		globalParams:              map[string]interface{}{},
+		flags: map[string]interface{}{},
 	}
 }
 
 func (fb *flagsBuilder) Build() map[string]interface{} {
 	flags := map[string]interface{}{}
-
-	if paramsAreNotEmpty(fb.containersDataParams) {
-		flags["containers"] = map[string]interface{}{
-			"manager": map[string]interface{}{
-				"configuration": map[string]interface{}{
-					"data": fb.containersDataParams,
-				},
-			},
-		}
-	}
-
-	if paramsAreNotEmpty(fb.globalParams) {
-		flags["global"] = fb.globalParams
+	for key, value := range fb.flags {
+		flagPath := strings.Split(key, ".")
+		appendFlag(flags, flagPath, value)
 	}
+	return flags
+}
 
-	if paramsAreNotEmpty(fb.webhookValuesParams) {
-		flags["webhook"] = map[string]interface{}{
-			"values": fb.webhookValuesParams,
+func appendFlag(flags map[string]interface{}, flagPath []string, value interface{}) {
+	currentFlag := flags
+	for i, path := range flagPath {
```

Comment:
IMO name `path` is hard to understand when near is `flagPath`. Maybe something like `flagPathElement`, `flagPathSection`, or shorter `pathElement`, ...

Reply:
`pathElement` looks elegant

Reply:
I propose `pathPart`